src/llama-mmap.cpp

```diff
@@ -471,14 +471,19 @@ struct llama_mlock::impl {
 
         char* errmsg = std::strerror(errno);
         bool suggest = (errno == ENOMEM);
-
+#if defined(GGML_VISIONOS) || defined(GGML_TVOS)
+        // visionOS/watchOS/tvOS dont't support RLIMIT_MEMLOCK
+        // Skip resource limit checks on visionOS/watchOS/tvOS
+        suggest = false;
+#else
```

Comment:
Instead of introducing `GGML_VISIONOS` and `GGML_TVOS`, use the target conditions like we do in `ggml.c`:

https://github.com/ggml-org/llama.cpp/blob/56d7a9f81274717fe035db192f315a049261c7fa/ggml/src/ggml.c#L65-L68

You just need to include `<TargetConditionals.h>` on `__APPLE__`.

Reply:
Nice, I was not aware of that. Thanks!